src/main/resources/lib/enonic/react4xp/serviceRoots.es6

```diff
@@ -1,52 +1,49 @@
 const portal = require('/lib/xp/portal');
 
-const initServiceUrlRoot = (serviceName, label) => {
-    const url = portal.serviceUrl({service: serviceName}) + '/';
-    return url;
-};
+const ROOT_URLS = {};
 
-let ASSET_ROOT_URL;
-const getAssetRoot = () => {
-    if (ASSET_ROOT_URL === undefined) {
-        ASSET_ROOT_URL = initServiceUrlRoot('react4xp', 'ASSET_ROOT_URL');
+const initServiceUrlRoot = (siteId, serviceName, key) => {
+    if (siteId === undefined) {
+        throw Error('siteId is undefined, cant getAssetRoot');
+    }
+    let rootUrlsForSite = ROOT_URLS[siteId];
+    if (rootUrlsForSite === undefined) {
+        ROOT_URLS[siteId] = {};
+        rootUrlsForSite = ROOT_URLS[siteId];
+    }
+    const existingUrl = rootUrlsForSite[key];
+    if (existingUrl !== undefined) {
+        return existingUrl;
     }
-    return ASSET_ROOT_URL;
+    const siteUrl = portal.pageUrl({id: siteId})
+    //const url = portal.serviceUrl({service: serviceName, type:'server'}) + '/';
+    const altUrl = (`${siteUrl}/_/service/${app.name}/${serviceName}/`).replace(/\/+/, '/');
+    ROOT_URLS[siteId][key] = altUrl;
+    return altUrl;
 };
 
-let CLIENT_ROOT_URL;
-const getClientRoot = () => {
-    if (CLIENT_ROOT_URL === undefined) {
-        CLIENT_ROOT_URL = initServiceUrlRoot('react4xp-client', 'CLIENT_ROOT_URL');
-    }
-    return CLIENT_ROOT_URL;
+const getAssetRoot = (siteId) => {
+    return initServiceUrlRoot(siteId,'react4xp', 'ASSET_ROOT_URL');
```

Comment:
'ASSET_ROOT_URL' looks like a constant that should be defined once and used multiple places?